test/xbgpu/test_engine.py

```diff
@@ -853,12 +855,15 @@ async def _send_data(
             for i, (corrprod_output, capture_stop_heap_index) in enumerate(
                 zip(corrprod_outputs, corrprod_capture_stop_heap_indices)
             ):
-                if capture_stop_heap_index > 0:
+                if capture_stop_heap_index is not None and capture_stop_heap_index > 0:
```

Comment:
It looks like you've introduced None to mean "don't stop" but left behind non-positive numbers to also mean "don't stop". Just eliminate the latter.